src/mtype.d

```diff
@@ -7140,6 +7140,11 @@ extern (C++) final class TypeDelegate : TypeNext
         }
         else if (ident == Id.funcptr)
         {
+            if (!(flag & 2) && sc.func && !sc.intypeof && sc.func.setUnsafe())
```

Comment:
`flag & 1` is borderline okay (odd/even) but clearly there's more structure here than `2` reveals. Could you please define some flags that clarify what `flag` is about?